pkg/sqlite/migrations/013_rm_truncated_deprecations.go

```diff
@@ -16,11 +16,11 @@ var rmTruncatedDeprecationsMigration = &Migration{
 	Id: RmTruncatedDeprecationsMigrationKey,
 	Up: func(ctx context.Context, tx *sql.Tx) error {
 
-		// Delete deprecation history for all bundles that no longer exist in the channel_entries table
+		// Delete deprecation history for all bundles that no longer exist in the operatorbundle table
```

Comment:
Why are we modifying an existing migration rather than adding a new migration?

Reply:
This change is solely for the purpose of keeping around entries in the deprecated table for deprecated channel heads. (otherwise, we have just an operatorbundle table entry for them). If we're adding another migration for this, it would be too late: the deprecated channel heads would already have been removed from everything but the operatorbundle table

Reply:
This migration is to ensure that deprecated channel head will not get dropped, which they would with the channel table losing the channel entry https://github.com/operator-framework/operator-registry/blob/29e90dea6a1c2ef12075ca866284cd8f267c2bba/pkg/sqlite/load.go#L1448
The channel head+channel_entry had to be dropped so the deprecated bundle wouldn't show up in console and the graph rebuilding from channel heads means any free entry in the channel entry table would be dropped anyway.

This method did bring up an interesting new bug: removing a channel head after deprecating it causes it to persist in the operatorbundle table even on package removal, preventing the bundle from ever being overwritten/added again

The migration should be safe though, we'd retain more deprecated entries if a migration was re-done, but it shouldn't affect any previously migrated index. Assuming no one was using 'deprecate channel head' as an equivalent for 'delete channel head'